h2o-core/src/main/java/water/api/schemas3/JobV3.java

```diff
@@ -91,8 +93,14 @@ public JobV3() {}
     msec = job.msec();
     ready_for_view = job.readyForView();
 
-    Keyed dest_type = (Keyed)TypeMap.theFreezable(job._typeid);
-    dest = job._result == null ? null : KeyV3.make(dest_type.makeSchema(),job._result);
+    Keyed dest_type;
+    Value value = DKV.get(job._result);
```

Comment:
IMHO this is not the best place to fix this. We are keeping stale info in the DKV and overriding it here.

Instead of that, we could deprecate the old information when a new Job is started with a result_key that overwrites an existing Job result.

Take a look here: https://github.com/h2oai/h2o-3/blob/master/h2o-core/src/main/java/water/Job.java#L286
- here we know we are running a new job that will overwrite the results of another job. This is where we know the old results will no longer be relevant, I think the older job shouldn't point to the new one anymore.

Reply:
i am not sure I understand what you mean, I tried something like this:

```java
    // 3 - Update job list
    final Key jobkey = _key;
    final Key jobResultKey = _result;
    new TAtomic<JobList>() {
      @Override public JobList atomic(JobList old) {
        if( old == null ) old = new JobList();
        Key[] jobs = old._jobs;
        for (Key<Job> jk : old._jobs) {
          Job j = DKV.get(jk).get();
          if (j._result.equals(jobResultKey)) {
            j._result = null;
          }
        }
        old._jobs = Arrays.copyOf(jobs, jobs.length + 1);
        old._jobs[jobs.length] = jobkey;
        return old;
      }
    }.invoke(LIST);
```

But Job#_result is final, so we need to duplicate and replace the old job instance when the result is owerwritten.

Reply:
Yes, this is what I was thinking. However, fetching from DKV in an atomic update is not a good idea. I think your original approach is fine. Thanks for trying it out